app/controllers/textings_controller.rb

```diff
@@ -0,0 +1,108 @@
+class TextingsController < ApplicationController
+
+  def create
+    # Make a request to Textellent API. If request successful we update our DB
+    response = post_textellent(get_db_data)
+
+    if response['status'] == 'success'
+      recipient = TextingRecipient.find_by(phone_number: texting_params[:phone_number])
+
+      if recipient
+        texting = Texting.new(
+          texting_recipient_id: recipient.id,
+          service_id: texting_params[:service_id]
+          )
+
+        if texting.save
+          render status: :ok, json: { message:'Message sent' }
+        else
+          render status: :bad_request
+        end
+
+      else
+        recipient = TextingRecipient.new(
+          recipient_name: texting_params[:recipient_name],
+          phone_number: texting_params[:phone_number]
+          )
+
+        recipient.save
+
+        texting = Texting.new(
+          texting_recipient_id: recipient.id,
+          service_id: texting_params[:service_id]
+          )
+
+        if texting.save
+          render status: :ok, json: { message:'Message sent' }
+        else
+          render status: :bad_request
+        end
```

Comment:
This branch of the nested `if/else` statements is very similar to the other branch where you create the `Texting` object. I think you could actually unify them into a single branch if you just pull out the conditional for when the TextingRecipient doesn't yet exist.

Here's a rough sketch of the change I'm suggesting, with the details omitted:

```rb
# The current code you have:

recipient = try_to_find_recipient()
if recipient
  texting = create_new_texting()
else
  recipient = create_new_recipient()
  texting = create_new_texting()
end


# Change that I am suggesting:
recipient = try_to_find_recipient()
if not recipient:
  recipient = create_new_recipient()

texting = create_new_texting()
```